### PR TITLE
doc: fix metrics doc

### LINKED ISSUE #505: metrics does not work
**What happened**:
follow page under: https://github.com/kubernetes-sigs/azuredisk-csi-driver/tree/master/deploy/example/metrics
```
# kubectl get svc csi-azuredisk-controller -n kube-system
NAME                       TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)     AGE
csi-azuredisk-controller   ClusterIP   10.0.230.247   <none>        29604/TCP   30s
root@andydev:~/go/src/sigs.k8s.io/azuredisk-csi-driver/deploy/example/metrics# curl http://10.0.230.247:29604/metrics | grep cloudprovider_azure
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:16 --:--:--     0^C
root@andydev:~/go/src/sigs.k8s.io/azuredisk-csi-driver/deploy/example/metrics# curl https://10.0.230.247:29604/metrics | grep cloudprovider_azure
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:03 --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:04 --:--:--     0

  0     0    0     0    0     0      0      0 --:--:--  0:00:20 --:--:--     0^C
```

**What you expected to happen**:

**How to reproduce it**:

**Anything else we need to know?**:

**Environment**:
- CSI Driver version:
- Kubernetes version (use `kubectl version`):
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/assign @ZeroMagic 
cc @nilo19 

may also add an e2e test to make sure it works in the future.

### PR DESCRIPTION
**Pull request by andyzhangx**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
doc: fix metrics doc

**Which issue(s) this PR fixes**:

Fixes #505

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by andyzhangx** [comment]
/assign @ZeroMagic
cc @nilo19

File: deploy/example/e2e_usage.md [hunk 1]
```diff
@@ -6,9 +6,9 @@
 kubectl create -f https://raw.githubusercontent.com/kubernetes-sigs/azuredisk-csi-driver/master/deploy/example/storageclass-azuredisk-csi.yaml
 ```
 
- - Create an azuredisk CSI PVC
+ - Create a statefulset with Azure Disk mount
 ```
-kubectl create -f https://raw.githubusercontent.com/kubernetes-sigs/azuredisk-csi-driver/master/deploy/example/pvc-azuredisk-csi.yaml
+kubectl create -f https://raw.githubusercontent.com/kubernetes-sigs/azuredisk-csi-driver/master/deploy/example/statefulset.yaml
```
Comment (ZeroMagic) [inline deploy/example/e2e_usage.md:11]: If using statefulset, it seems that the other two parts need to be rewrote.

Comment (andyzhangx) [reply deploy/example/e2e_usage.md:11]: which part? For static provisioning, we still use `pvc-azuredisk-csi.yaml`. And for dynamic provisioning, use statefulset example is easier to demo.

Comment (andyzhangx) [reply deploy/example/e2e_usage.md:11]: let's merge this PR first, will modify the content in another PR.

Comment (ZeroMagic) [reply deploy/example/e2e_usage.md:11]: 2. validate PVC status and create an nginx pod
3. enter the pod container to do validation

**Review by ZeroMagic — APPROVED** [review]
/lgtm

**Pull request merged**